components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/NotificationTemplateManagerImpl.java

```diff
@@ -0,0 +1,720 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.email.mgt;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.email.mgt.constants.TemplateMgtConstants;
+import org.wso2.carbon.email.mgt.internal.I18nMgtDataHolder;
+import org.wso2.carbon.email.mgt.store.TemplatePersistenceManager;
+import org.wso2.carbon.email.mgt.store.TemplatePersistenceManagerFactory;
+import org.wso2.carbon.email.mgt.util.I18nEmailUtil;
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementServerException;
+import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerClientException;
+import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerException;
+import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerInternalException;
+import org.wso2.carbon.identity.governance.exceptions.notiification.NotificationTemplateManagerServerException;
+import org.wso2.carbon.identity.governance.model.NotificationTemplate;
+import org.wso2.carbon.identity.governance.service.notification.NotificationChannels;
+import org.wso2.carbon.identity.governance.service.notification.NotificationTemplateManager;
+import org.wso2.carbon.identity.organization.management.service.OrganizationManager;
+import org.wso2.carbon.identity.organization.management.service.exception.OrganizationManagementException;
+import org.wso2.carbon.identity.organization.management.service.util.OrganizationManagementUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.email.mgt.constants.TemplateMgtConstants.DEFAULT_EMAIL_NOTIFICATION_LOCALE;
+import static org.wso2.carbon.email.mgt.constants.TemplateMgtConstants.DEFAULT_SMS_NOTIFICATION_LOCALE;
+import static org.wso2.carbon.email.mgt.util.I18nEmailUtil.normalizeLocaleFormat;
+
+/**
+ * Provides functionality to manage notification templates.
+ */
+public class NotificationTemplateManagerImpl implements NotificationTemplateManager {
+
+    private static final Log log = LogFactory.getLog(NotificationTemplateManagerImpl.class);
+
+    private final TemplatePersistenceManager userDefinedTemplatePersistenceManager;
+    private final TemplatePersistenceManager systemTemplatePersistenceManager;
+    private final TemplatePersistenceManager unifiedTemplatePersistenceManager;
```

Comment:
Rename to a general name : eg : templatePersistenceManager

Reply:
addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/250/commits/4a4dae818d530081e8a91a33d6d70da13c375c9c